//
// Copyright 2022 DMetaSoul
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.dmetasoul.metaspore.serving";
option java_outer_classname = "MetaSporeServingProto";
option objc_class_prefix = "SPORE";
option cc_enable_arenas = true;

package metaspore.serving;

service Predict {
  rpc Predict(PredictRequest) returns (PredictReply) {}
}

message PredictRequest {
  string                model_name = 1;
  map<string, string>   parameters = 3;
  map<string, bytes>    payload    = 5;
}

message PredictReply {
  map<string, bytes>    payload    = 1;
  map<string, string>   extras     = 3;
}

message LoadRequest {
  string model_name = 1;
  string version = 3;
  string dir_path = 5;
}

message LoadReply {
  string msg = 1;
}

service Load {
  rpc Load(LoadRequest) returns (LoadReply) {}
}